### ISSUE TITLE
"Not in VTerm mode" error when pressing M-r for history search

### ISSUE THREAD

**Issue by gondalez** [syl20bnr/spacemacs#14100]
#### Thanks 🙏 
Thank you @thanhvg and @smile13241324 for a477ea63ced532ee1771b6c18af44d809c30c384, I can't wait to use this feature! I hope you can help me out with this bug/error.

#### Description :octocat:
Pressing `M-r` to search history in VTerm gives "Not in VTerm mode" error.

#### Reproduction guide :beetle:
- Start Emacs
- set `(shell :variables spacemacs-vterm-history-file-location "~/.zsh_history")` (I use zsh)
- Open terminal in vterm mode 
- Press `M-r`

*Observed behaviour:* :eyes: :broken_heart:
- "not in VTerm mode" message

See screenshot, where I also printed the `mode-name` variable:
![image](https://user-images.githubusercontent.com/1468141/97766133-1941ac00-1b50-11eb-9baa-9aa52169bfae.png)

This assertion seems to be getting tripped: https://github.com/syl20bnr/spacemacs/blob/a477ea63ced532ee1771b6c18af44d809c30c384/layers/%2Btools/shell/funcs.el#L247

*Expected behaviour:* :heart: :smile:
- helm history search 😌 

#### System Info :computer:
- OS: darwin
- Emacs: 27.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 0291e9eb5)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(emacs-lisp helm org
            (shell :variables shell-default-height 30 shell-default-position 'bottom shell-default-shell 'vterm spacemacs-vterm-history-file-location "~/.zsh_history")
            version-control treemacs html yaml javascript react prettier
            (ruby :variables ruby-test-runner 'rspec)
            cucumber auto-completion git github markdown org syntax-checking terraform emoji templates)
```
- System configuration features: NOTIFY KQUEUE ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS MODULES THREADS JSON PDUMPER LCMS2 GMP

**Comment by duianto** [comment]
The following PRs changes have been applied to the `develop` branch:
fix #14100 #14101

@gondalez, can you test that this worked as expected, if it did then you can close this issue.

**Comment by gondalez** [comment]
Confirmed fixed, thanks ✅
